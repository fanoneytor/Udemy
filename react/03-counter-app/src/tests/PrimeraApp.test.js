import React from 'react';
import '@testing-library/jest-dom';
import { render } from "@testing-library/react";
import PrimeraApp from "../PrimeraApp";
import { shallow } from 'enzyme';

describe('Pruebas en <PrimeraApp />', () => {
    
    /*test('debe de mostrar el mensaje "Hola soy Stefano"', () => {

        const saludo = 'Hola soy Stefano';

        const { getByText } = render( <PrimeraApp saludo = { saludo }/>);

        expect( getByText( saludo )).toBeInTheDocument();

    })*/

    test('debe de mostrar <PrimeraAPP /> correctamente', () => {
        const saludo = "Hola soy Stefano";
        const wrapper = shallow( <PrimeraApp saludo={ saludo}/> )

        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de mostrar el subtitulo enviado por props', () => {
        const saludo = 'Hola soy Stefano';
        const subtitulo = 'Soy un subtitulo';
        const wrapper = shallow( 
            <PrimeraApp 
                saludo={ saludo }
                subtitulo = { subtitulo }
            />
        );

        const textoParrafo = wrapper.find('p').text();
        
        expect( textoParrafo ).toBe( subtitulo );
        
    })
})