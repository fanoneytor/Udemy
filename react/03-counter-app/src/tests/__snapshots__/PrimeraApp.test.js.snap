// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> debe de mostrar <PrimeraAPP /> correctamente 1`] = `
<Fragment>
  <h1>
     
    Hola soy Stefano
     
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;

exports[`Pruebas en <PrimeraApp /> debe de mostrar el subtitulo enviado por props 1`] = `
<Fragment>
  <h1>
     
    Hola soy Stefano
     
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
